cmd/common-methods.go

```diff
@@ -105,7 +106,9 @@ func putTargetStream(alias string, urlStr string, reader io.Reader, size int64,
 	if err != nil {
 		return 0, err.Trace(alias, urlStr)
 	}
-	n, err := targetClnt.Put(reader, size, metadata, progress)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
We can combine this `cancel` with `trapCh` so that if we do get a signal like CTRL-C `<-trapCh` we cleanly exit.

Reply:
done.